### PR TITLE
too specific instructions for Knights Travails

### LINKED ISSUE #26301: <JavaScript>: too specific instructions for Knights Travails
### Describe your suggestion

This is related to this [discord discussion](https://discord.com/channels/505093832157691914/514204569572605952/1148248156165898250).

1. assignment instructs learners to create a Board and Knight factories/classes.
This isn't needed to solve KT and it doesn't relate to the previous lessons either, I found that confusing and unnecessary.

2.assignment instructs learners to use a specific data structure, that is trees.
it says: "you’ll need to use a data structure that’s similar (but not identical) to a binary tree."

This isn't needed either, but it's related to previous lessons, so I understand that it could be a deliberate choice to restrict learners to use trees for this specific problem.

If it is deliberate then it would be more appropriate to reword  the above quoted instructions.
And if it isn't then the instructions would need to be changed a little.

Here is an [example of a project](https://github.com/OmarNaguib/Knights-Travails) that doesn't create a board or a knight, and doesn't use trees either.

image

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-knights-travails

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### (Optional) Discord Name

Omar (onsmak) @onsmak

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by OmarNaguib**
## Because

-Wasn't clear that there is no need for GUI
-instructions suggested users have to make Board and Knight objects
-instructions suggested users have to represent the problem as a tree

## This PR

-Clarified there is no need for GUI
-used more generalized instructions

## Issue

Closes #26301

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: javascript/computer_science/project_knights_travails.md [hunk 1]
```diff
@@ -24,8 +25,8 @@ You can think of the board as having 2-dimensional coordinates.  Your function w
   * `knightMoves([0,0],[7,7]) == [[0,0],[2,1],[4,2],[6,3],[4,4],[6,5],[7,7]]` or `knightMoves([0,0],[7,7]) == [[0,0],[2,1],[4,2],[6,3],[7,5],[5,6],[7,7]]`
 
 <div class="lesson-content__panel" markdown="1">
-1. Put together a script that creates a game board and a knight.
-2. Treat all possible moves the knight could make as children in a tree.  Don't allow any moves to go off the board.
+1. Think about the rules of the board and knight, make sure to follow them.
```
Comment (JoshDevHub) [inline javascript/computer_science/project_knights_travails.md:28]:
```suggestion
1. Think about the rules of the board and knight, and make sure to follow them.
```

File: javascript/computer_science/project_knights_travails.md [hunk 2]
```diff
@@ -24,8 +25,8 @@ You can think of the board as having 2-dimensional coordinates.  Your function w
...
+2. For every square there is a number of possible moves, choose a data structure that will allow you to work with them .  Don't allow any moves to go off the board.
```
Comment (JoshDevHub) [inline javascript/computer_science/project_knights_travails.md:29]:
```suggestion
2. For every square there is a number of possible moves, choose a data structure that will allow you to work with them. Don't allow any moves to go off the board.
```

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
Great stuff. Just a couple small fixes

Also, would you be comfortable making these same changes for the ruby path's version of this project? The projects have the exact same instructions, and the only differences are things like using `snake_case` with Ruby and `camelCase` with JavaScript. That's not relevant for the changes you're making here though. You should be able to just make these same changes over in the ruby course's `project_knight_travails.md` file.

**Comment by JoshDevHub** [comment]
Do you want to make the same changes in the ruby course @OmarNaguib ?

If not that's fine, I just wanted to give you the opportunity to add that in before merging this.

**Comment by OmarNaguib** [comment]
@JoshDevHub, Thank you for the opportunity, I think i added the required changes

**Review by JoshDevHub — APPROVED** [review]
Looks great. Thank you for contributing @OmarNaguib :raised_hands:

**Pull request merged**